src/utils/command-invocations.js

```diff
@@ -23,19 +21,27 @@ const maybeLogin = context => {
   });
 };
 
+const hasRegisteredApp = async () => {
+  try {
+    const app = await getLinkedAppConfig();
+    return Boolean(app.id);
+  } catch (err) {
+    return false;
+  }
+};
+
 // check if the user needs to be registered in and do it if so
 // returns a promise
-const maybeRegisterApp = context => {
-  if (!fileExistsSync(constants.CURRENT_APP_FILE)) {
+const maybeRegisterApp = async context => {
+  if (!await hasRegisteredApp()) {
     context.line(
       "Looks like this is your first push. Let's register your app on Zapier."
     );
-    return getInput('Enter app title (Ctrl-C to cancel):\n\n  ').then(title =>
-      register(context, title, { printWhenDone: false })
+    return await getInput('Enter app title (Ctrl-C to cancel):\n\n  ').then(
```

Comment:
minor, seems weird to return the promise chain here. Why not

```js
const input = await getInput('Enter' // ...
await register(context, // ...
return null
```

Also, i'm not sure we need the return statement at all.

Reply:
I totally missed it! Cleaned up in 4830272.